core/src/test/scala/org/apache/spark/sql/delta/DeltaOptionSuite.scala

```diff
@@ -240,4 +241,24 @@ class DeltaOptionSuite extends QueryTest
       }
     }
   }
+
+  test("partitionOverwriteMode is set to invalid value") {
+    withTempDir { tempDir =>
+      val invalidMode = "ADAPTIVE"
+      val e = intercept[IllegalArgumentException] {
+        Seq(1, 2, 3).toDF
+          .withColumn("part", $"value" % 2)
+          .write
+          .format("delta")
+          .partitionBy("part")
+          .option("partitionOverwriteMode", invalidMode)
+          .save(tempDir.getAbsolutePath)
+      }
+      assert(e.getMessage ===
```

Comment:
mhh its actually hard to check. something like this doesn't work in tests:

`withSQLConf(SQLConf.PARTITION_OVERWRITE_MODE.key -> invalidMode) {`

because it errors out immediately. SQLConf will not accept this invalid value.

Reply:
oh cool. thats good then. spark ensures that this scenario does not need to be explicitly tested